app/src/main/java/com/safetyheads/akademiaandroida/Adapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.safetyheads.akademiaandroida
+
+import android.content.Context
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.safetyheads.akademiaandroida.databinding.CardBinding
+
+
+class Adapter(private val list: List<DataListActivites>, private val context: Context) :
+    RecyclerView.Adapter<Adapter.MyViewHolder>() {
+
+    inner class MyViewHolder(binding: CardBinding) : ViewHolder(binding.root) {
+        val textViewCard = binding.textViewCard
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val CardBinding = CardBinding.inflate(inflater, parent, false)
+        return MyViewHolder(CardBinding)
+    }
+
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        val item = list[position].name
+        holder.textViewCard.text = list[position].name
+
+        holder.itemView.setOnClickListener {
+            if (item != "com.safetyheads.akademiaandroida.ListActivity") {
```

Comment:
Hardcodowanie nazwy klasy z pakietem to nie jest najlepsze rozwiązanie. W momencie gdy ktoś zmieni nazwę klasy lub przeniesie do pakietu to ten kod już nie zadziała.